mifospay/src/main/java/org/mifospay/qr/ui/ReadQrScreen.kt

```diff
@@ -0,0 +1,237 @@
+package org.mifospay.qr.ui
+
+import android.Manifest
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import android.os.Build
+import android.util.Size
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.view.PreviewView
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.content.ContextCompat
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.mifospay.R
+import org.mifospay.core.designsystem.component.MfLoadingWheel
+import org.mifospay.core.designsystem.component.MifosScaffold
+import org.mifospay.core.designsystem.component.PermissionBox
+import org.mifospay.core.designsystem.icon.MifosIcons
+import org.mifospay.core.ui.EmptyContentScreen
+import org.mifospay.qr.presenter.ReadQrUiState
+import org.mifospay.qr.presenter.ReadQrViewModel
+import org.mifospay.theme.MifosTheme
+import org.mifospay.utils.QrCodeAnalyzer
+
+@Composable
+fun ShowQrScreenRoute(
+    viewModel: ReadQrViewModel = hiltViewModel(),
+    backPress: () -> Unit
+) {
+    val uiState = viewModel.readQrUiState.collectAsStateWithLifecycle()
+
+    ReadQrScreen(
+        uiState = uiState.value,
+        backPress = backPress,
+        scanQR = { viewModel.scanQr(it) }
+    )
+}
+
+@Composable
+fun ReadQrScreen(
+    uiState: ReadQrUiState,
+    backPress: () -> Unit,
+    scanQR: (Bitmap) -> Unit,
+) {
+    var isFlashOn by remember { mutableStateOf(false) }
```

Comment:
use `rememberSaveable`? state change such as screen rotation will turn off the flash!

Reply:
done